src/org/zalando/stups/mint/worker/external/bucket_storage.clj

```diff
@@ -0,0 +1,14 @@
+(ns org.zalando.stups.mint.worker.external.bucket_storage)
+
+(defmacro StorageException
+  [msg data]
+  `(ex-info ~msg (merge ~data {:type "StorageException"})))
+
+(defn storage-exception? [e]
+  (= "StorageException" (:type (ex-data e))))
+
+(defprotocol BucketStorage
```

Comment:
`defprotocol` seems like the Java way to solve the problem. What do you think about [defmulti](https://clojuredocs.org/clojure.core/defmulti)? E.g. the function could be dispatched based on the bucket name.